Makefile

```diff
@@ -8,7 +8,7 @@ fmt:
 
 lint: vendor
 	@echo "✓ Linting source code with https://staticcheck.io/ ..."
-	@staticcheck ./...
+	@go run honnef.co/go/tools/cmd/staticcheck@v0.4.0 ./...
```

Comment:
Was running into versioning error where `min` and `max` keywords were not being recognized (these were added as of Go `1.21`) so matched this line to [Terraform provider](https://github.com/databricks/terraform-provider-databricks/blob/master/Makefile#L15) to ensure the right version of go is used when running `staticcheck`.

<img width="610" alt="Screenshot 2023-10-08 at 5 14 47 PM" src="https://github.com/databricks/cli/assets/87999496/2529cc58-3886-4715-80e1-f098c9b1af18">

Reply:
@arpitjasa-db which go version are you using?

Reply:
Tried `1.21.0` and then updated to `1.21.2` but still running into the same issue even though `min` and `max` keyword support was added as of `Go 1.21.0`.

Reply:
To debug, can you run:
```
$ staticcheck --version
staticcheck 2023.1.3 (v0.4.3)
```

This one works for me (installed through `go install honnef.co/go/tools/cmd/staticcheck@latest`).

I prefer we don't lock to an earlier version because we get additional checks latest.

Could you revert the changes to the Makefile entirely?

Reply:
Odd I have the latest version but it still doesn't work
```
(base) arpit.jasapara@C02FX2U3MD6R cli % staticcheck --version
staticcheck 2023.1.6 (v0.4.6)
```
But agreed on not pinning to an earlier version, reverted the change